osquery/tables/networking/windows/routes.cpp

```diff
@@ -0,0 +1,178 @@
+
+  if (result != NO_ERROR) {
+    FreeMibTable(ipTable);
+
+    return results;
+  }
+
+  auto numEntries = ipTable[0]->NumEntries;
+  auto interfaces = getInterfaceRowMapping();
+  auto adapters = getAdapterAddressMapping();
+
+  for (unsigned long i = 0; i < numEntries; ++i) {
+    Row r;
+    std::string interfaceIpAddress;
+    PVOID ipAddress = nullptr;
+    PVOID gateway = nullptr;
+    auto currentRow = ipTable[0]->Table[i];
+    auto addrFamily = currentRow.DestinationPrefix.Prefix.si_family;
+    auto actualInterface = interfaces.at(currentRow.InterfaceIndex);
+    if (addrFamily == AF_INET6) {
+      r["mtu"] = INTEGER(actualInterface.NlMtu);
+      // These are all technically "on-link" addresses according to
+      // `route print -6`.
+      r["type"] = "local";
+      ipAddress = reinterpret_cast<struct sockaddr_in6*>(
+          &currentRow.DestinationPrefix.Prefix.Ipv6.sin6_addr);
+      gateway = reinterpret_cast<struct sockaddr_in6*>(
+          &currentRow.NextHop.Ipv6.sin6_addr);
+    } else if (addrFamily == AF_INET) {
+      ipAddress = reinterpret_cast<struct sockaddr_in*>(
+          &currentRow.DestinationPrefix.Prefix.Ipv4.sin_addr);
+      gateway = reinterpret_cast<struct sockaddr_in*>(
+          &currentRow.NextHop.Ipv4.sin_addr);
+
+      // The software loopback is not returned by GetAdaptersInfo, so any
+      // lookups into that index must be skipped and default values set.
+      PIP_ADAPTER_INFO actualAdapter = nullptr;
+      if (currentRow.InterfaceIndex != 1) {
+        actualAdapter = adapters.at(currentRow.InterfaceIndex).get();
+        interfaceIpAddress = actualAdapter->IpAddressList.IpAddress.String;
+        r["mtu"] = INTEGER(actualInterface.NlMtu);
+      } else {
+        interfaceIpAddress = "127.0.0.1";
+        r["mtu"] = UNSIGNED_BIGINT(0xFFFFFFFF);
+      }
+      r["type"] = currentRow.Loopback ? "local" : "remote";
+    }
+    std::vector<char> buffer(INET6_ADDRSTRLEN);
+    InetNtop(addrFamily, ipAddress, buffer.data(), buffer.size());
+    r["destination"] = SQL_TEXT(buffer.data());
+    InetNtop(addrFamily, gateway, buffer.data(), buffer.size());
+    r["gateway"] = SQL_TEXT(buffer.data());
+    r["interface"] = SQL_TEXT(interfaceIpAddress);
+    r["metric"] = INTEGER(currentRow.Metric + actualInterface.Metric);
+    r["netmask"] =
+        SQL_TEXT(std::to_string(currentRow.DestinationPrefix.PrefixLength));
+    // TODO: implement routes flags
+    r["flags"] = SQL_TEXT("-1");
+
+    results.push_back(r);
+
+    // Cleanup
+    SecureZeroMemory(ipAddress, sizeof(ipAddress));
+    SecureZeroMemory(gateway, sizeof(gateway));
+    buffer.clear();
+
+    ipAddress = nullptr;
+    gateway = nullptr;
+  }
+  FreeMibTable(ipTable);
+  interfaces.clear();
+
+  return results;
+}
+
+QueryData genRoutes(QueryContext& context) {
+  QueryData results;
+  QueryData routes = genIPRoutes(context);
+
+  for (auto const& item : routes) {
+    results.push_back(item);
```

Comment:
Seems like this might be unnecessary? Why not just have `QueryData routes = genIPRoutes(context);` and `return results`?